angular/src/lib/date-range-picker/date-range-input.component.ts

```diff
@@ -0,0 +1,149 @@
+import { Component, Input, ViewChild, ElementRef, Output, EventEmitter, OnInit } from '@angular/core';
+import { DatePickerService } from '../date-picker/date-picker.service';
+
+
+@Component({
+  selector: 'md-date-range-input',
+  template: `
+    <label class="md-label" for="dateRangeInput">{{labelValue}}</label>
+    <div class="md-input__icon-container">
+      <input type="text"
+        #dateRangeInput
+        class="md-input dirty"
+        tabIndex="0"
+        [value]="getValue()"
+        id="dateRangeInput"
+        name="dateRangeInput"
+        [placeholder]="placeholder"
+        (focus)='handleFocus($event)'
+        (blur)='handleBlur($event)'
+        (keydown)='onInputPress($event)'
+        (mouseover)='handleMouseover()'
+        (mouseout)='handleMouseout($event)'
+        readonly />
+      <button #closeBtn
+        type="button"
+        (focus)='stop($event)'
+        (click)='clear($event)'
+        style="display:none;"
+        class="md-button md-button--36 md-button--icon"
+        alt="clear input"
+        type="button"
+        tabIndex="-1">
+        <span #closeSpan class="md-button__children">
+          <i #closeI class="icon icon-clear-active_16"></i>
+        </span>
+      </button>
+    </div>
+  `,
+  host: {
+    class: 'md-input-group small-5 columns'
+  }
+})
+export class DateRangeInputComponent implements OnInit {
+  private _allowDay: string;
+  /** @prop the value of the label */
```

Comment:
Missing default

Reply:
`const key = event.key || event.which || event.keyCode; `
the value of the key and keyCode is different. Will we add another switch structor with the different values?

Reply:
Delete _allowDay since it is not used

Reply:
will use '@angular/cdk/keycodes' now to rewrite the event